test/fixtures/main/expected.json

```diff
@@ -2,11 +2,17 @@
   {
     "id": "foo.bar.baz",
     "description": "The default message",
-    "defaultMessage": "Hello World!"
+    "defaultMessage": "Hello World!",
+    "files": [
+      "test/fixtures/main/actual.js"
+    ]
   },
   {
     "id": "foo.bar.biff",
     "description": "Another message",
-    "defaultMessage": "Hello Nurse!"
+    "defaultMessage": "Hello Nurse!",
+    "files": [
```

Comment:
How can i use this information? Anyway, you shouldn't override default behavior.
So, if you need this information in result file, you should specify option for this with default value, that don't change behavior.

Reply:
> How can i use this information?
The only use case I see is being able to remove unused messages from a file. Although, since this lib is an aggregator, it doesn't sound wrong to collect the files.

> Anyway, you shouldn't override default behavior.
> So, if you need this information in result file, you should specify option for this with default value, that don't change behavior.

So you mean that adding an attribute to a descriptor is a breaking change and that's why I'm changing the default behavior? If so, I can add `files` attribute only when `cleanUpUnusedMessages` is enabled. But again, keep in mind that since it's an aggregator, it is a nice intended default behavior to have the files collected.

Also, it's important to note that we won't necessarily have a full scan. When using `devServer` and/or hot reload, it's common to have a single file scan, and then removing an unused descriptor for each file.

Replying to the first comment:

> Perhaps I do not understand the idea, but for me it looks like this:
> 
> * We go through all the files and retrieve messages
> * For each message, add information about the file where it is declared
> * And then we delete those messages for which they were not announced.

yep, it's exactly like you described.

> Probably, if you make more clear tests, i will understand your idea.

Sure, I'll give a try. Do you have a suggestion?

Reply:
> Sure, I'll give a try. Do you have a suggestion?

Test will be right, if you add unused message.
test/fixtures/main/cleared-expected.json should remain unchanged.